components/kubeflow/katib-launcher/src/launch_experiment.py

```diff
@@ -99,11 +100,13 @@ def main(argv=None):
   config.load_incluster_config()
   api_client = k8s_client.ApiClient()
   experiment = Experiment(version=args.version, client=api_client)
+  exp_name = str(args.name)+'-'+str(uuid.uuid4().hex)
+  
   inst = {
     "apiVersion": "%s/%s" % (ExperimentGroup, args.version),
     "kind": "Experiment",
     "metadata": {
-      "name": args.name,
+      "name": exp_name,
```

Comment:
I think names might be limited to 63 characters max.

Reply:
Yeah, the current implementation would leave 31 characters for user.
Would you advise on appending a smaller UUID instead of 32 character long?

Reply:
I think the name should be explicitly truncated. For example, `exp_name[0:63]`.
BTW, what do you think about just using generated names?
For example you can have 
```
"generateName": sanitize(arg.name[0:31]) + '-',
```
Or even use fully generated names.

Reply:
yes, actually generateName is better!
I tested generateName with a Job template and it works okay with _kubectl create_ command but not _apply_ command which probably makes sense because resource has not been created yet! 
And I checked this code is creating the resource in **experiment.create(**) so it should work okay with this too.

Reply:
but actually, args.name is also being used in wait_for_condition() and delete() methods and I am not sure how it would work if the name is randomly generated at the execution time.

Reply:
The return value from `create` might have the actual name.

P.S. I'm OK with keeping your name generation, but I'd like to prevent the length from going over the limit.

Reply:
Updated it so that the experiment name is limited to 64 characters.